storage/engine/rocksdb.go

```diff
@@ -166,6 +167,40 @@ func (r *RocksDB) getInternal(key proto.EncodedKey, snapshotHandle *C.DBSnapshot
 	return cStringToGoBytes(result), nil
 }
 
+// GetProto fetches the value at the specified key and unmarshals it.
+func (r *RocksDB) GetProto(key proto.EncodedKey, msg gogoproto.Message) (
+	ok bool, keyBytes, valBytes int64, err error) {
+	return r.getProtoInternal(key, msg, nil)
+}
+
+func (r *RocksDB) getProtoInternal(key proto.EncodedKey, msg gogoproto.Message,
+	snapshotHandle *C.DBSnapshot) (ok bool, keyBytes, valBytes int64, err error) {
+	if len(key) == 0 {
+		err = emptyKeyError()
+		return
+	}
+	var result C.DBString
+	if err = statusToError(C.DBGet(r.rdb, snapshotHandle, goToCSlice(key), &result)); err != nil {
+		return
+	}
+	if result.len <= 0 {
+		return
+	}
+	ok = true
+	if msg != nil {
+		// Make a byte slice that is backed by result.data. This slice
+		// cannot live past the lifetime of this method, but we're only
+		// using it to unmarshal the proto.
+		// data := C.GoBytes(unsafe.Pointer(result.data), result.len)
+		data := (*[0x7fffffff]byte)(unsafe.Pointer(result.data))[:result.len:result.len]
```

Comment:
Can you pull this out into a function? The 0x7fffffff functions as a 2GB limit on the size of data we can support, right? Is there any way we can avoid a magic constant here? Or make it `2**63-1` instead of `2**31-1`?